app/components/post_list/post/footer/footer.tsx

```diff
@@ -0,0 +1,178 @@
+            ...followingButtonContainerBase,
+            paddingLeft: 0,
+        },
+        notFollowing: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Heading', 75),
+        },
+        followingButtonContainer: {
+            ...followingButtonContainerBase,
+            backgroundColor: changeOpacity(theme.buttonBg, 0.08),
+            borderRadius: 4,
+        },
+        following: {
+            color: theme.buttonBg,
+            ...typography('Heading', 75),
+        },
+        followSeparator: {
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.16),
+            height: 16,
+            marginRight: 12,
+            width: 1,
+        },
+    };
+});
+
+const Footer = ({currentUserId, participants, serverUrl, teamId, teammateNameDisplay, testID, theme, thread}: Props) => {
+    const styles = getStyleSheet(theme);
+    const onUnfollow = useCallback(preventDoubleTap(() => {
+        updateThreadFollowing(serverUrl, teamId, thread.id, false);
+    }), []);
+
+    const onFollow = useCallback(preventDoubleTap(() => {
+        updateThreadFollowing(serverUrl, teamId, thread.id, true);
+    }), []);
+
+    let repliesComponent;
+    let followButton;
+    if (thread.replyCount) {
+        repliesComponent = (
+            <>
+                <View style={styles.replyIconContainer}>
+                    <CompassIcon
+                        name='reply-outline'
+                        size={18}
+                        color={changeOpacity(theme.centerChannelColor, 0.64)}
+                    />
+                </View>
+                <FormattedText
+                    style={styles.replies}
+                    testID={`${testID}.reply_count`}
+                    id='threads.replies'
+                    defaultMessage='{count} {count, plural, one {reply} other {replies}}'
+                    values={{
+                        count: thread.replyCount,
+                    }}
+                />
+            </>
+        );
+    }
+    if (thread.isFollowing) {
+        followButton = (
+            <TouchableOpacity
+                onPress={onUnfollow}
+                style={styles.followingButtonContainer}
+                testID={`${testID}.following`}
+            >
+                <FormattedText
+                    id='threads.following'
+                    defaultMessage='Following'
+                    style={styles.following}
+                />
+            </TouchableOpacity>
+        );
+    } else {
+        followButton = (
+            <>
+                <View style={styles.followSeparator}/>
+                <TouchableOpacity
+                    onPress={onFollow}
+                    style={styles.notFollowingButtonContainer}
+                    testID={`${testID}.follow`}
+                >
+                    <FormattedText
+                        id='threads.follow'
+                        defaultMessage='Follow'
```

Comment:
This is just a nit - so optional.

I see that the structure of `followButton` is the same. We can make use of an object that will hold the values in the isFollowing and notfollowing and then based on the boolean `thread.isFollowing`, you feed in the info to that structure.